### PR TITLE
John conroy/query portal logs

### LINKED ISSUE #2328: Faster log searching
GOAL: We want to find all the errors (and warnings, probably?) in the last week in the production logs. John has visibility on the logs through AWS Cloudwatch, but it seems slow: How can we quickly get a short list?

The [docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/SearchDataFilterPattern.html) just have this to say:
> If you have a lot of log data, search might take a long time to complete. To speed up a search, you can do the following:
> - If you are using the AWS CLI, you can limit the search to just the log streams you are interested in. For example, if your log group has 1000 log streams, but you just want to see three log streams that you know are relevant, you can use the AWS CLI to limit your search to only those three log streams within the log group.
> - Use a shorter, more granular time range, which reduces the amount of data to be searched and speeds up the query.

If that's not sufficient, maybe we need to bring in an index... but we're not dealing with so much traffic that the vanilla solution is obviously inadequate.

Also possible that we're logging too much: I don't think that's the case, but maybe we should be doing something to partition the logs?

Finally, once you've got something that's runs faster, I think it'd be good to reduce it to the CLI: easier to say 'copy and paste this' than explaining how to click through a UI.

### PR DESCRIPTION
**Pull request by john-conroy**
Fix #2328. I created an IAM role one can assume to run the script and right now I'm the only one who has the access to assume that role. I think that's fine for now, but if you would like access let me know.

It seems like we're exceeding the response size limit ( at least for the last week). I might bump it up to 500.

### CODE REVIEW

File: query-portal-logs.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    return {d['field']: d['value']}
+
+
+if __name__ == "__main__":
+    client = boto3.Session(profile_name='harvarddev').client('logs')
+
+    query = client.start_query(
+        logGroupName='portal-ui-logs',
+        startTime=int((datetime.now() - timedelta(days=7)).timestamp()),
+        endTime=int(datetime.now().timestamp()),
```
Comment (mccalluc) [inline query-portal-logs.py:19]: Is this required, or could you just leave it off?

**Review by mccalluc — CHANGES_REQUESTED** [review]
Thanks! Some python tweaks.

File: query-portal-logs.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    query_id = query['queryId']
+
+    response = None
+
+    while response is None or response['status'] == 'Running':
+        print('Waiting for query to complete')
+        time.sleep(5)
+        response = client.get_query_results(
+            queryId=query_id
+        )
```
[the remark this answers is not in the document]
Comment (john-conroy) [reply query-portal-logs.py:36]: Hmm.. the client has a [get_waiter method](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/logs.html#CloudWatchLogs.Client.get_waiter), but `boto3.client('logs').waiter_names` returns an empty list.

[the remark this answers is not in the document]
Comment (john-conroy) [reply query-portal-logs.py:36]: Yeah I looked through the`botocore` repo as well and there aren't waiters defined for logs.

**Review by mccalluc — APPROVED** [review]

File: query-portal-logs.py (hunk 1 above)
Comment (john-conroy) [reply query-portal-logs.py:17]: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/logs.html#CloudWatchLogs.Client.start_query `endTime` is required.

**Pull request merged**